spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/support/InvalidQueryConfiguration.java

```diff
@@ -0,0 +1,19 @@
+package org.springframework.data.jdbc.repository.support;
+
+import org.springframework.data.jdbc.repository.query.Query;
+
+/**
+ * Exception thrown when both {@link Query#resultSetExtractorClass()} and {@link Query#rowMapperClass()} are used in one {@link Query}.
+ *
+ * @author Evgeni Dimitrov
+ */
+public class InvalidQueryConfiguration extends RuntimeException {
```

Comment:
We should rather reuse existing exception types instead of introducing new ones. We could use `InvalidDataAccessApiUsageException` in this case.

Another thing here is that we derive data-access-related exceptions from `DataAccessException`.